instrumentation/github.com/99designs/gqlgen/otelgqlgen/gqlgen.go

```diff
@@ -0,0 +1,162 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgqlgen
+
+import (
+	"context"
+	"os"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+
+	otelcontrib "go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	oteltrace "go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName         = "go.opentelemetry.io/contrib/instrumentation/github.com/99designs/gqlgen/otelgqlgen"
+	extensionName      = "OpenTelemetry"
+	complexityLimit    = "ComplexityLimit"
+	defaultServiceName = "GraphQLService"
+)
+
+type Tracer struct {
+	serviceName             string
+	complexityExtensionName string
+	tracer                  oteltrace.Tracer
+}
+
+var _ interface {
+	graphql.HandlerExtension
+	graphql.ResponseInterceptor
+	graphql.FieldInterceptor
+} = Tracer{}
+
+func (a Tracer) ExtensionName() string {
+	return extensionName
+}
+
+func (a Tracer) Validate(_ graphql.ExecutableSchema) error {
+	return nil
+}
+
+func (a Tracer) InterceptResponse(ctx context.Context, next graphql.ResponseHandler) *graphql.Response {
+	ctx, span := a.tracer.Start(ctx, operationName(ctx), oteltrace.WithSpanKind(oteltrace.SpanKindServer))
+	defer span.End()
+	if !span.IsRecording() {
+		return next(ctx)
+	}
+
+	oc := graphql.GetOperationContext(ctx)
+
+	serviceName := os.Getenv("OTEL_SERVICE_NAME")
+	if serviceName == "" {
+		serviceName = defaultServiceName
+	}
+	span.SetAttributes(
+		semconv.ServiceNameKey.String(serviceName),
```

Comment:
this should be removed.
the resource detector is responsible for it
see:
- https://github.com/open-telemetry/opentelemetry-go/blob/b8561785c01cbcefa88d6fabe33d774869fcd637/sdk/resource/env.go
- https://opentelemetry.io/docs/go/exporting_data/#resources

Reply:
oh, sorry for misunderstood. fixed now :)